radix-engine-tests/tests/blueprints/component/src/component.rs

```diff
@@ -53,3 +61,88 @@ mod component_test {
         }
     }
 }
+
+#[derive(Debug, PartialEq, Eq, ScryptoSbor, NonFungibleData)]
+pub struct ComponentTest2NonFungible {
+    pub val: String,
+}
+
+#[blueprint]
+mod component_test2 {
+
+    struct ComponentTest2 {
+        vault: Vault,
+    }
+
+    impl ComponentTest2 {
+        pub fn create_component() -> Global<ComponentTest2> {
+            let resource_manager = ResourceBuilder::new_integer_non_fungible::<
+                ComponentTest2NonFungible,
+            >(OwnerRole::None)
+            .mint_roles(mint_roles! {
+                minter => rule!(allow_all);
+                minter_updater => rule!(deny_all);
+            })
+            .burn_roles(burn_roles! {
+                burner => rule!(allow_all);
+                burner_updater => rule!(deny_all);
+            })
+            .create_with_no_initial_supply();
+
+            let vault = resource_manager.create_empty_vault();
+
+            Self { vault }
+                .instantiate()
+                .prepare_to_globalize(OwnerRole::None)
+                .globalize()
+        }
+
+        pub fn generate_nft(&mut self) -> Bucket {
+            let resource_manager = self.vault.resource_manager();
+
+            let id = NonFungibleLocalId::integer(1);
+
+            let bucket = resource_manager.mint_non_fungible(
+                &id,
+                ComponentTest2NonFungible {
+                    val: String::from("FirstNFT"),
+                },
+            );
+            self.vault.put(bucket);
+            let bucket = self.vault.as_non_fungible().take_non_fungible(&id);
+            bucket.into()
+        }
+
+        pub fn generate_nft_proof(&mut self) -> (Bucket, Proof) {
+            let bucket = self.generate_nft();
+            let proof = bucket.create_proof_of_all();
+            (bucket, proof)
+        }
+    }
+}
+
+#[blueprint]
+mod component_test3 {
+
+    struct ComponentTest3 {
+        resource_id: ResourceAddress,
+    }
+
+    impl ComponentTest3 {
+        pub fn create_component(resource_id: ResourceAddress) -> Global<ComponentTest3> {
+            Self { resource_id }
+                .instantiate()
+                .prepare_to_globalize(OwnerRole::None)
+                .globalize()
+        }
+
+        pub fn check_proof(&self, proof: Proof) {
+            proof.check(self.resource_id).drop();
+        }
+
+        pub fn check_proof_and_burn_bucket(&self, bucket: Bucket, proof: Proof) {
+            proof.check(self.resource_id).drop();
+            bucket.burn();
```

Comment:
Can we also try a test where the `bucket.burn()` happens before `proof.check()`?

Reply:
I checked that before and there is an error: `KernelError(CallFrameError(DropNodeError(NodeBorrowed(NodeId("f8eb898e760f64a79e74ac60a8465bb2c6cd7a68aae9bbe7eb70947d59d1")))))`

Reply:
Oh okay great, can we atleast add that test (with the error check)